user-management/src/main/java/de/tum/in/www1/artemis/usermanagement/security/ArtemisInternalAuthenticationProvider.java

```diff
@@ -0,0 +1,91 @@
+package de.tum.in.www1.artemis.usermanagement.security;
+
+import de.tum.in.www1.artemis.domain.User;
+import de.tum.in.www1.artemis.repository.UserRepository;
+import de.tum.in.www1.artemis.service.connectors.ConnectorHealth;
+import de.tum.in.www1.artemis.service.user.PasswordService;
+import de.tum.in.www1.artemis.usermanagement.service.user.UserCreationService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.security.authentication.InternalAuthenticationServiceException;
+import org.springframework.security.core.Authentication;
+import org.springframework.stereotype.Component;
+
+import java.util.Optional;
+
+@Component
+@ConditionalOnProperty(value = "artemis.user-management.use-external", havingValue = "false")
+public class ArtemisInternalAuthenticationProvider extends ArtemisAuthenticationProviderImpl implements ArtemisAuthenticationProvider {
+
+    private final Logger log = LoggerFactory.getLogger(ArtemisInternalAuthenticationProvider.class);
+
+    public ArtemisInternalAuthenticationProvider(UserRepository userRepository, PasswordService passwordService, UserCreationService userCreationService) {
+        super(userRepository, passwordService, userCreationService);
+    }
+
+    @Override
+    public User getOrCreateUser(Authentication authentication, String firstName, String lastName, String email, boolean skipPasswordCheck) {
+        final var password = authentication.getCredentials().toString();
+        final var optionalUser = userRepository.findOneByLogin(authentication.getName().toLowerCase());
+        final User user;
+        if (optionalUser.isEmpty()) {
+            user = userCreationService.createInternalUser(authentication.getName(), password, null, firstName, lastName, email, null, null, "en");
+        }
+        else {
+            user = optionalUser.get();
+            if (!skipPasswordCheck) {
+                final var storedPassword = passwordService.decryptPassword(user);
+                if (!password.equals(storedPassword)) {
+                    throw new InternalAuthenticationServiceException("Authentication failed for user " + user.getLogin());
+                }
+            }
+        }
+
+        return user;
+    }
+
+    @Override
+    public void addUserToGroup(User user, String group) {
+        // nothing to do, this was already done by the UserService, this method is only needed when external management is active
+    }
+
+    @Override
+    public void removeUserFromGroup(User user, String group) {
+        // nothing to do, this was already done by the UserService, this method is only needed when external management is active
+    }
+
+    @Override
+    public void createUserInExternalUserManagement(User user) {
+        // This should not be invoked. As we only use internal management, nothing needs to be done in case it is invoked.
```

Comment:
[nit] If this should not be invoked, can we maybe add an assert/throw an exception (such that it crashes and we get a sentry report) or maybe log in case it is invoked?

Reply:
I added UnsupportedOperationException to the method